content/articles/create-a-weather-app-in-node.js-using-openweathermap-api/index.md

```diff
@@ -0,0 +1,560 @@
+
+```javascript
+    // Request for data using the URL
+    request(url, function(err, response, body) {
+
+        // On return, check the json data fetched
+        if (err) {
+            res.render('index', { weather: null, error: 'Error, please try again' });
+        } else {
+            let weather = JSON.parse(body);
+```
+
+- We shall check if our weather data returned is undefined. This will indicate errors. If not, we shall proceed to store the content.
+
+```javascript
+            // We shall output it in the console just to make sure that the data being displayed is what we want
+            console.log(weather);
+
+            if (weather.main == undefined) {
+                res.render('index', { weather: null, error: 'Error, please try again' });
+            } else {
+                // we shall use the data got to set up our output
+                let place = `${weather.name}, ${weather.sys.country}`,
+                    /* We shall calculate the current timezone using the data fetched*/
+                    weatherTimezone = `${new Date(weather.dt * 1000 - (weather.timezone * 1000))}`;
+                let weatherTemp = `${weather.main.temp}`,
+                    weatherPressure = `${weather.main.pressure}`,
+                    /* We shall fetch the weather icon and its size using the icon data*/
+                    weatherIcon = `http://openweathermap.org/img/wn/${weather.weather[0].icon}@2x.png`,
+                    weatherDescription = `${weather.weather[0].description}`,
+                    humidity = `${weather.main.humidity}`,
+                    clouds = `${weather.clouds.all}`,
+                    visibility = `${weather.visibility}`,
+                    main = `${weather.weather[0].main}`,
+                    weatherFahrenheit;
+                weatherFahrenheit = ((weatherTemp * 9 / 5) + 32);
+
+                // We shall also round off the value of the degrees fahrenheit calculated into two decimal places
+                function roundToTwo(num) {
+                    return +(Math.round(num + "e+2") + "e-2");
+                }
+                weatherFahrenheit = roundToTwo(weatherFahrenheit);
+```
+
+- The stored values shall now be rendered unto the webpage to generate a static webpage which shall be displayed.
+
+```javascript
+                // We shall now render the data to our page (index.ejs) before displaying it out
+                res.render('index', { weather: weather, place: place, temp: weatherTemp, pressure: weatherPressure, icon: weatherIcon, description: weatherDescription, timezone: weatherTimezone, humidity: humidity, fahrenheit: weatherFahrenheit, clouds: clouds, visibility: visibility, main: main, error: null });
+            }
+        }
+    });
+});
+```
+
+> **Note:** You can fetch and display much data as you will need in your project as needed in the json received.
+
+- Set up our port configurations
```

Comment:
At this step, you are doing more than setting up port configs, you are also starting the server and setting it's listening port. Be sure to add that to the article.

Reply:
Sure

Reply:
Done!